src/python/bindings/bind.cpp

```diff
@@ -83,5 +84,30 @@ PYBIND11_MODULE(smartredisPy, m) {
         .def("get_meta_scalars", &PyDataset::get_meta_scalars)
         .def("get_meta_strings", &PyDataset::get_meta_strings)
         .def("get_name", &PyDataset::get_name);
+
+    // Register exception classes
+    py::register_exception_translator([](std::exception_ptr p) {
```

Comment:
I think in general, these need to somehow map to ``Exception`` derived classes we define in Python.

A few ways to do this

1. bind python exceptions to the bind layer and bind directly. 
2. throw, catch, and throw a better python error. (this is more fragile imo, because the mapping has to be right)

Reply:
Deferred to Python error handling ticket.  First approach is the one that will be taken btw,